test/unit/go.mod

```diff
@@ -0,0 +1,78 @@
+module github.com/aws/cluster-api-provider-cloudstack-staging/test/unit
```

Comment:
> Why do we need a separate module for unit tests?
It removes the dependencies from our main module which helps with tools like Dep Check. It also breaks the test code out into its own directory structure. Our testing is becoming more and more extensive, with subdirectories and fixtures and dummies and whatnot. At this point since there's _so_ much of it all, it's clearer overall if the two are separate.

> And I'm not sure if "staging" is the right term for it if we do

Yes, just an artifact of the draft status of the PR.